OpenRA.Game/MiniYaml.cs

```diff
@@ -437,34 +437,40 @@ static List<MiniYamlNode> MergePartial(List<MiniYamlNode> existingNodes, List<Mi
 				return existingNodes;
 
 			var ret = new List<MiniYamlNode>(existingNodes.Count + overrideNodes.Count);
+			var plainKeys = new HashSet<string>(existingNodes.Count + overrideNodes.Count);
+
+			foreach (var node in existingNodes)
+				MergeNode(node);
+			foreach (var node in overrideNodes)
+				MergeNode(node);
 
-			foreach (var node in existingNodes.Concat(overrideNodes))
+			void MergeNode(MiniYamlNode node)
 			{
 				// Append Removal nodes to the result.
-				// Therefore: we know the remainder of the loop deals with plain nodes.
+				// Therefore: we know the remainder of the method deals with plain nodes.
```

Comment:
This comment doesn't really make sense anymore, since we `return` out of this method and will be a bit confusing if you don't know this was a loop before.

Reply:
Adjusted to `.. remainder of the method deals with a plain node.`